### ISSUE TITLE
Built-in flyouts position blocks with x and y values incorrectly

### ISSUE THREAD

**Issue by BeksOmega** [google/blockly#6280]
**Describe the bug**

If you define a toolbox where the block info has x and y properties, the blocks will be positioned incorrectly.

**To Reproduce**

1. Create a toolbox definition.
2. Add blocks to it
3. Give those blocks x and y properties (That are non-zero)
4. Observe how they are positioned incorrectly.

**Expected behavior**

The toolbox should ignore the x and y properties on blocks because it has its own gap/separator logic for positioning blocks.

**Additional context**

- The [BlockInfo type](https://github.com/google/blockly/blob/fa58cbaaf7456ab227c2214af5dd0e7025dc731c/core/utils/toolbox.ts#L19) should be updated to not include X and Y.
- Blockly has two kinds of flyouts built-in, horizontal and vertical. You should update the layout logic in both of them.
- The flyouts should be changed to use [moveTo](https://github.com/google/blockly/blob/fa58cbaaf7456ab227c2214af5dd0e7025dc731c/core/block_svg.ts#L414) [instead of moveBy](https://github.com/google/blockly/blob/61322294da70c770b32abb28b9d695af75f61a16/core/flyout_vertical.js#L250)

**Tips for Getting Started**
We have a lot of information about how to get started [here](https://developers.google.com/blockly/guides/contribute/get-started) including details about our commit message style and how to open a PR.

**Comment by Samridhi-98** [comment]
Hey @BeksOmega, I would like to look into this issue.
Will you please explain the issue more and where to start?

**Comment by varshneydevansh** [comment]
Hi! @maribethb just wanted to know is this issue or the [issue](https://github.com/google/blockly-samples/issues/1264) where this is referenced are still pending? Would you mind if I took on this task?

**Comment by maribethb** [comment]
Hi @varshneydevansh either of these issues are still open & available! Let me know if you'd like to work on them. I'll add some additional pointers to the first comment here in this issue.

**Comment by varshneydevansh** [comment]
Hi @maribethb 

I'd like to work on these. Yesterday, I set up the working environment by following the [document](https://developers.google.com/blockly/guides/contribute/get-started) and learned how blockly works.

**Comment by varshneydevansh** [comment]
Which file should I look for, currently I am looking at – `toolbox.ts`, `flyout_base.ts`, `block_svg.ts`

**Comment by varshneydevansh** [comment]
@BeksOmega I think these are the files which are related to the problem.

I removed the `X` and `Y` from the block info type from the file `core/utils/toolbox.ts`

Now, I have to replace the `moveBy` definition carefully to use `moveTo` in the files – `core/flyout_horizontal.ts` and `core/flyout_vertical.ts`

**Comment by BeksOmega** [comment]
This code is moving the block back to `(0, 0)`. So if you're positioning them using absolute coordinates (`moveTo`) rather than relative coordinates (`moveBy`) you should be able to delete that call to `moveBy` in `flyout_base` =)

I hope that helps @varshneydevansh !

**Comment by varshneydevansh** [comment]
I wasn't able to create the PR at that time now before creating the PR I just looked again at the updated code and wanted to know which `moveTo` should I use?

I came across to this just now after noticing in the `core/flyout_button.ts` - 

https://github.com/varshneydevansh/blockly/blob/e147ab30c64f1fa740f329e20e20d5c2ac035e6f/core/flyout_button.ts#L224

I have used this `moveTo` - https://github.com/varshneydevansh/blockly/blob/e147ab30c64f1fa740f329e20e20d5c2ac035e6f/core/block_svg.ts

**Comment by BeksOmega** [comment]
The block `moveTo` is correct!

**Comment by varshneydevansh** [comment]
I am actually a little lost with testing my changes. How do I correctly do it, and how do I write the Unit tests for the changes?

The right side one is with no changes in the code, and the left one is with the changes made.
![image](https://github.com/google/blockly/assets/20411128/db20b713-4494-40e2-a4d1-25bdf2285828)

**Comment by varshneydevansh** [comment]
So, do look back again how this happened?

**Comment by BeksOmega** [comment]
Yeah if you want to take another look @varshneydevansh that would be great =)
